### ISSUE TITLE
declare a "port" parameter for os_security_group_rule

### ISSUE THREAD

**Issue by ansibot** [ansible/ansible#29575]
From @jgbillings on 2016-10-17T23:10:02Z
##### ISSUE TYPE
- Feature Idea
##### COMPONENT NAME

os_security_group_rule

I end up with a lot of "os_security_group_rule:" statements that have: 

port_range_max: "{{ item }}"
port_range_min: "{{ item }}"

For rules where only a single port is needed.   Is there some reason not to define a "port:" parameter that if that has been passed set port_range_min and port_range_max to that value? 

That should pass the right things down the chain through shade to openstack.    

Copied from original issue: ansible/ansible-modules-core#5289

**Comment by ansibot** [comment]
From @ansibot on 2016-10-17T23:10:02Z

@jgbillings we are unable to validate that "" is an existing ansible module.

We are expecting a component name that matches something that can be found in:
- http://docs.ansible.com/ansible/list_of_all_modules.html

Please revise the component name in the description.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
From @ansibot on 2016-10-17T23:10:02Z

@jgbillings, we need more information to resolve this issue. For that reason, we put this Issue into the 'needs_info' state.

When you have answered the questions or filled in the missing data, please comment with text 'ready_for_review' and we will put this Issue back into review.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
From @jgbillings on 2016-10-17T23:10:02Z

ready_for_review

**Comment by ansibot** [comment]
From @ansibot on 2016-10-17T23:10:02Z

@emonty, @shrews, @juliakreger, @j2sol, @rcarrillocruz, ping. This issue is waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
From @j2sol on 2016-10-17T23:10:02Z

Adding more input options does increase the complexity of the module, and the documentation. Given that it's trivially easy to specify the same port for the minimum and the maximum, I'm not inclined to want to add another way to input those values.

**Comment by ansibot** [comment]
From @jgbillings on 2016-10-17T23:10:02Z

What if say only minimum was set, could that be interpreted to mean that only a single port required and set that same port for both minimum and maximum?

**Comment by ansibot** [comment]
From @j2sol on 2016-10-17T23:10:02Z

I'm not really inclined to do that either. That's non-obvious behavior. What's the reticence to listing both the min/max ? This is what maps to the API itself.

**Comment by ansibot** [comment]
From @jgbillings on 2016-10-17T23:10:02Z

Just seems redundant, for example in one config I have: 

```
        port_range_min: 8140
        port_range_max: 8140

        port_range_min: 25
        port_range_max: 25

        port_range_min: 25826
        port_range_max: 25826

        port_range_min: 5000
        port_range_max: 5000
```

Why do in eight lines what you can do with four?

**Comment by ansibot** [comment]
From @Thingee on 2016-10-17T23:10:02Z

I hear what @j2sol is saying. All interfaces give some way of doing this, and this is the modules way of doing things. It's also good to see we have this covered in our [documentation example](http://docs.ansible.com/ansible/os_security_group_rule_module.html#examples) already.

Horizon for example shows a single port field or the ability to select port range. OpenStackClient allows both a single port to be passed or "min:max" to be passed in.

http://docs.openstack.org/developer/python-openstackclient/command-objects/security-group-rule.html#security-group-rule-create

<img width="715" alt="screen shot 2016-11-02 at 3 11 15 pm" src="https://cloud.githubusercontent.com/assets/47583/19949411/af011bec-a10e-11e6-83e9-9c35fe0c0f51.png">

**Comment by ansibot** [comment]
From @j2sol on 2016-10-17T23:10:02Z

If I understand it right, @Thingee you are advocating for adding a single port option to the module?

I wouldn't block it being added, I just don't like "fixing" UX issues of a given API in all the various downstream consumers, instead of just fixing the UX issue directly. But that's one hell of a windmill to tilt against.

**Comment by ansibot** [comment]
From @Thingee on 2016-10-17T23:10:02Z

I actually wasn't advocating for the change. More just say each interface has their own way of dealing with the problem.

**Comment by ansibot** [comment]
cc @bennojoy @dagnello @dharmabumstead @thingee
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by biomassives** [comment]
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  acmeideal@gmail.com
    Domain biomassiv.es has exceeded the max emails per hour (103/100 (103%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
Received: from github-smtp2-ext4.iad.github.net ([192.30.252.195]:35888 helo=github-smtp2a-ext-cp1-prd.iad.github.net)
	by chi-server32.websitehostserver.net with esmtps (TLSv1.2:ECDHE-RSA-AES256-GCM-SHA384:256)
	(Exim 4.89)
	(envelope-from <noreply@github.com>)
	id 1drSrj-002R4q-EE
	for greg@biomassiv.es; Mon, 11 Sep 2017 12:51:23 -0500
Date: Mon, 11 Sep 2017 10:50:43 -0700
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=github.com;
	s=pf2014; t=1505152243;
	bh=N6WNbJL71QvHHDh/CWewcG+CaYfAdceuvSVw7c7kvt8=;
	h=From:Reply-To:To:Cc:In-Reply-To:References:Subject:List-ID:
	 List-Archive:List-Post:List-Unsubscribe:From;
	b=wW8hVYI9yy6/O0AzXyOIFNaB0p6D9A+Be7pFlm8uJE3Ga/0kMtq2fyZOg3wsix3z4
	 3QYJ/5vEdMU6/RxfPsqoDgSidLx9i6J0lvFjYkpwMzfSHhufnvUoys9VF0rn1OZoDH
	 yK7K8wHwxtd0hX5Njk2+BD5HTkcetZ4xhN1z4z98=

**Comment by ansibot** [comment]
cc @omgjlk
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
cc @mnaser
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by mnaser** [comment]
wontfix

I think it's better that we stick to mapping to what the API wants to make sense.  Horizon just presents things in a different way that aim to be more GUI centric.  The comments seem to follow.
